app/src/lib/stores/app-store.ts

```diff
@@ -2211,11 +2211,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
     if (eligibleBranches.length >= FastForwardBranchesThreshold) {
       log.info(
-        `skipping fast-forward work because there are ${
+        `skipping fast-forward for all branches as there are ${
           eligibleBranches.length
         } local branches - this will run again when there are less than ${FastForwardBranchesThreshold} local branches tracking remotes`
       )
-      return
+
+      const defaultBranch = state.branchesState.defaultBranch
```

Comment:
This looks like the fifth reason a branch would be eligible for an update. Can we add that the default branch is always eligible to be fast-forwarded as a comment?

Reply:
:thumbsup: this also reminds me I need to do the same checks in here to ensure I'm not currently on `master`:

https://github.com/desktop/desktop/blob/00669f788655ac4606ee8c4d61e46e3a75f62e1b/app/src/lib/stores/app-store.ts#L2204-L2210

Reply:
Master